File: Dataset/Leetcode/train/5/85.cpp
 class Solution {
public:
    string XXX(string s) {
        int n = s.size();
        string ret;
        vector<vector<bool>> dp(n, vector<bool>(n));

        for(int len = 1; len <= n; ++len){
            for(int i = 0; i <= n - len; ++i){
                int j = i + len - 1;
                if(len == 1) dp[i][j] = true;
                else if(len == 2) dp[i][j] = (s[i] == s[j]);
                else dp[i][j] = (s[i] == s[j] && dp[i+1][j-1]);
                if((j - i + 1 > ret.size() && dp[i][j])) ret = s.substr(i, len);
            }
        }
        return ret;
    }
};

File: Dataset/Leetcode/train/9/85.java
class Solution {
    public boolean XXX(int x) {
        String str = Integer.toString(x);
        int len = str.length();
        for(int i=len-1;i>0;i--){
            if(str.charAt(i)!=str.charAt(len-i-1))    return false;
            if(i<=len-i-1)    break;
        }
        return true;
    }
}

